packages/zendesk-support-adapter/src/filters/user.ts

```diff
@@ -0,0 +1,236 @@
+  })
+}
+
+const mergeUserReplacers = (replacers: UserReplacer[]): UserReplacer => (instance, mapping) => {
+  replacers.forEach(replacer => { replacer(instance, mapping) })
+}
+
+const DEFAULT_REPLACER_PARAMS_FOR_ACTIONS = [{ fieldName: 'actions', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'follower' }] }]
+const DEFAULT_REPLACER_PARAMS_FOR_CONDITIONS = [
+  { fieldName: 'conditions.all', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }] },
+  { fieldName: 'conditions.any', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }] },
+]
+
+const TYPE_NAME_TO_REPLACER: Record<string, UserReplacer> = {
+  automation: replaceConditionsAndActionsCreator([
+    ...[{ fieldName: 'actions', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'follower' }, { name: 'notification_user', valuePath: 'value.0' }] }],
+    ...DEFAULT_REPLACER_PARAMS_FOR_CONDITIONS,
+  ]),
+  macro: mergeUserReplacers([
+    replaceConditionsAndActionsCreator(DEFAULT_REPLACER_PARAMS_FOR_ACTIONS),
+    replaceRestriction,
+  ]),
+  routing_attribute_value: replaceConditionsAndActionsCreator([
+    { fieldName: 'conditions.all', fieldsToReplace: [{ name: 'requester_id' }] },
+    { fieldName: 'conditions.any', fieldsToReplace: [{ name: 'requester_id' }] },
+  ]),
+  sla_policy: replaceConditionsAndActionsCreator([
+    { fieldName: 'filter.all', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }] },
+    { fieldName: 'filter.any', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }] },
+  ], true),
+  trigger: replaceConditionsAndActionsCreator([
+    ...[{ fieldName: 'actions', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'follower' }, { name: 'notification_user', valuePath: 'value.0' }, { name: 'notification_sms_user', valuePath: 'value.0' }] }],
+    ...[
+      { fieldName: 'conditions.all', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }, { name: 'role' }] },
+      { fieldName: 'conditions.any', fieldsToReplace: [{ name: 'assignee_id' }, { name: 'requester_id' }, { name: 'role' }] },
+    ],
+  ]),
+  view: mergeUserReplacers([
+    replaceConditionsAndActionsCreator(DEFAULT_REPLACER_PARAMS_FOR_CONDITIONS),
+    replaceRestriction,
+  ]),
+  workspace: workspaceReplacer,
+}
+
+const getUsers = async (paginator: clientUtils.Paginator): Promise<User[]> => {
+  const paginationArgs = {
+    url: '/users?role[]=admin&role[]=agent',
+    paginationField: 'next_page',
+  }
+  const users = (await toArrayAsync(
+    paginator(paginationArgs, page => makeArray(page) as clientUtils.ResponseValue[])
+  )).flat().flatMap(response => response.users)
+  if (!areUsers(users)) {
+    return []
+  }
+  return users
+}
+
+const thereAreRelevantChangesToReplace = (changes: Change<InstanceElement>[]): boolean => {
+  const changesTypeNames = _.uniq(changes.map(getChangeData).map(e => e.elemID.typeName))
+  return !_.isEmpty(_.intersection(changesTypeNames, Object.keys(TYPE_NAME_TO_REPLACER)))
```

Comment:
why calculate the full intersection just to find if there is at least one relevant type?
```suggestion
  return changes
    .map(getChangeData)
    .map(e => e.elemID.typeName)
    .some(typeName => Object.keys(TYPE_NAME_TO_REPLACER).includes(typeName))
```

Reply:
We don't need any of that